lib/instrumentation/modules/handlebars.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+var debug = require('debug')('elastic-apm')
+var shimmer = require('../shimmer')
+
+module.exports = function (handlebars, agent, version) {
+  debug('shimming handlebars.compile')
+  shimmer.wrap(handlebars, 'compile', wrapCompile)
+
+  return handlebars
+
+  function wrapTemplate (original) {
+    return function wrappedTemplate (data) {
+      var trace = agent.buildTrace()
+      var id = trace && trace.transaction.id
+
+      debug('intercepted call to handlebars render %o', {
+        id: id,
+        data: data
+      })
+
+      if (trace) trace.start('handlebars', 'template.handlebars.render')
+      var ret = original.call(this, data)
+      if (trace) trace.end()
+
+      return ret
+    }
+  }
+
+  function wrapCompile (original) {
+    return function wrappedCompile (input, options, hb) {
+      var trace = agent.buildTrace()
+      var id = trace && trace.transaction.id
+
+      debug('intercepted call to handlebars compile %o', {
+        id: id,
+        input: input
+      })
+
+      if (trace) trace.start('handlebars', 'template.handlebars.compile')
+      var ret = original.call(this, input, options, hb)
```

Comment:
There's a risk that the handlebars module at some point change their method signature for instance adding a 4th argument. If possible I like to just `original.apply(this, arguments)` to ensure that we pass on everything exactly as it was given to us

Reply:
fixed.